pinot-common/src/test/java/org/apache/pinot/common/metrics/AbstractMetricsTest.java

```diff
@@ -268,4 +274,157 @@ public void testMeteredMetrics() {
+    int gaugeOperationsRuntimeMs = 100;
+
+    ExecutorService executorService = Executors.newFixedThreadPool(2);
+    long endTime = System.currentTimeMillis() + gaugeOperationsRuntimeMs;
+
+    Future<?> addFuture = executorService.submit(() -> {
+      while (System.currentTimeMillis() < endTime + gaugeOperationsRuntimeMs) {
+        controllerMetrics.addValueToGlobalGauge(ControllerGauge.VERSION, 1L);
+      }
+    });
+    Future<?> removeFuture = executorService.submit(() -> {
+      while (System.currentTimeMillis() < endTime) {
+        controllerMetrics.removeGauge(ControllerGauge.VERSION.getGaugeName());
+      }
+    });
+
+    addFuture.get();
+    removeFuture.get();
+
+    executorService.shutdown();
+    boolean terminated = executorService.awaitTermination(1, TimeUnit.SECONDS);
+    Assert.assertTrue(terminated, "Tasks should complete and executor should shut down after 1 seconds.");
+    Assert.assertFalse(controllerMetrics.getMetricsRegistry().allMetrics().isEmpty());
+
+    Long gaugeValue = controllerMetrics.getGaugeValue(ControllerGauge.VERSION.getGaugeName());
+    Assert.assertNotNull(gaugeValue);
+    Assert.assertTrue(gaugeValue > 0);
+  }
+
+  @Test
+  public void testTableGaugeMetricsAsyncAddRemove() throws ExecutionException, InterruptedException {
+    ControllerMetrics controllerMetrics = buildTestMetrics();
+    int gaugeOperationsRuntimeMs = 100;
+
+    ExecutorService executorService = Executors.newFixedThreadPool(2);
+    long endTime = System.currentTimeMillis() + gaugeOperationsRuntimeMs;
+
+    Future<?> addFuture = executorService.submit(() -> {
+      while (System.currentTimeMillis() < endTime + gaugeOperationsRuntimeMs) {
+        controllerMetrics.addValueToTableGauge("test_table", ControllerGauge.VERSION, 1L);
+      }
+    });
+    Future<?> removeFuture = executorService.submit(() -> {
+      while (System.currentTimeMillis() < endTime) {
+        controllerMetrics.removeTableGauge("test_table", ControllerGauge.VERSION);
+      }
+    });
+
+    addFuture.get();
+    removeFuture.get();
+
+    executorService.shutdown();
+    boolean terminated = executorService.awaitTermination(1, TimeUnit.SECONDS);
+    Assert.assertTrue(terminated, "Tasks should complete and executor should shut down after 1 seconds.");
+    Assert.assertFalse(controllerMetrics.getMetricsRegistry().allMetrics().isEmpty());
+
+    Long gaugeValue = controllerMetrics.getGaugeValue(ControllerGauge.VERSION.getGaugeName() + ".test_table");
+    Assert.assertNotNull(gaugeValue);
+    Assert.assertTrue(gaugeValue > 0);
+  }
+
+  @Test
+  public void testInitializeGlobalMeters() {
+    ControllerMetrics controllerMetrics = buildTestMetrics();
+
+    controllerMetrics.initializeGlobalMeters();
+    Assert.assertFalse(controllerMetrics.getMetricsRegistry().allMetrics().isEmpty());
+    Assert.assertEquals(0, controllerMetrics.getGaugeValue(ControllerGauge.VERSION.getGaugeName()));
```

Comment:
nit: Here we can loop through controllerMetrics.getMeters() and filter out the ones where `isGlobal` is true to validate all global meters will be init to 0.

Reply:
Good catch, addressed.